lib/Models/Cesium.js

```diff
@@ -421,28 +430,84 @@ Cesium.prototype.zoomTo = function(viewOrExtent, flightDurationSeconds) {
                 });
             });
 
-        } else if (defined(viewOrExtent.position)) {
+        } else if (defined(target.entities)) {
+            // Zooming to a DataSource
+            if (target.isLoading && defined(target.loadingEvent)) {
+                var deferred = when.defer();
+                var removeEvent = target.loadingEvent.addEventListener(function() {
+                    removeEvent();
+                    deferred.resolve();
+                });
+                return deferred.promise.then(function() {
+                    return zoomToDataSource(that, target, flightDurationSeconds);
+                });
+            }
+            return zoomToDataSource(that, target);
+        } else if (defined(target.readyPromise)) {
+            return target.readyPromise.then(function() {
+                if (defined(target.boundingSphere)) {
+                    zoomToBoundingSphere(that, target, flightDurationSeconds);
+                }
+            });
+        } else if (defined(target.boundingSphere)) {
+            return zoomToBoundingSphere(that, target);
+        } else if (defined(target.position)) {
             that.scene.camera.flyTo({
                 duration: flightDurationSeconds,
-                destination: viewOrExtent.position,
+                destination: target.position,
                 orientation: {
-                    direction: viewOrExtent.direction,
-                    up: viewOrExtent.up
+                    direction: target.direction,
+                    up: target.up
                 }
             });
         } else {
             that.scene.camera.flyTo({
                 duration: flightDurationSeconds,
-                destination: viewOrExtent.rectangle
+                destination: target.rectangle
             });
         }
 
     }).then(function() {
         that.notifyRepaintRequired();
     });
-
 };
 
+var boundingSphereScratch = new BoundingSphere();
+
+function zoomToDataSource(cesium, target, flightDurationSeconds) {
+    return pollToPromise(function() {
+        var dataSourceDisplay = cesium.dataSourceDisplay;
+        var entities = target.entities.values;
+
+        var boundingSpheres = [];
+        for (var i = 0, len = entities.length; i < len; i++) {
+            var state = dataSourceDisplay.getBoundingSphere(entities[i], false, boundingSphereScratch);
+
+            if (state === BoundingSphereState.PENDING) {
+                return false;
+            } else if (state !== BoundingSphereState.FAILED) {
+                boundingSpheres.push(BoundingSphere.clone(boundingSphereScratch));
+            }
+        }
+
+        var boundingSphere = BoundingSphere.fromBoundingSpheres(boundingSpheres);
+        cesium.scene.camera.flyToBoundingSphere(boundingSphere, {
+            duration : flightDurationSeconds
+        });
```

Comment:
This is my fault, but this `pollToPromise` isn't going to work right because the function never returns `true`! So it will just repeat this thing every 100ms until it times out 5 seconds later. I think it just needs a `return true` after the `flyToBoundingSphere`.